api4/cloud.go

```diff
@@ -55,6 +55,8 @@ func (api *API) InitCloud() {
 
 	// GET /api/v4/cloud/cws-health-check
 	api.BaseRoutes.Cloud.Handle("/check-cws-connection", api.APIHandler(handleCheckCWSConnection)).Methods("GET")
+
+	api.BaseRoutes.Cloud.Handle("/delete-workspace", api.APISessionRequired(selfServeDeleteWorkspace)).Methods(http.MethodDelete)
```

Comment:
Heads up that we'll need agreement between this implementation and the frontend on HTTP method (see [comment here](https://github.com/mattermost/mattermost-webapp/pull/12077#discussion_r1096036961))

Reply:
Went with Delete. No need to change the code here.